lib/matplotlib/axes/_base.py

```diff
@@ -263,6 +263,14 @@ def _plot_args(self, tup, kwargs):
             raise ValueError('third arg must be a format string')
         else:
             linestyle, marker, color = None, None, None
+
+        # check that n values of None are passed in
```

Comment:
I want to leave the comment as I started to solve this by forcing the input to be float-like, but that would break datetime support (and I suspect bunches of the unit-aware code).